relay-general/src/store/transactions/processor.rs

```diff
@@ -184,23 +182,34 @@ impl<'r> TransactionsProcessor<'r> {
     }
 
     fn normalize_transaction_name(&self, event: &mut Event) -> ProcessingResult {
-        if matches!(
-            event.get_transaction_source(),
-            &TransactionSource::Url | &TransactionSource::Sanitized
-        ) {
+        let source = event
+            .transaction_info
+            .value()
+            .and_then(|i| i.source.value());
+
+        // Treat the transaction a a URL in the following cases:
+        // 1. It is marked with source:url
+        // 2. It is marked with source:sanitized, in which case we run normalization again.
+        // 3. It has no source attribute because it's from an old SDK version,
+        //    but it contains slashes and we expect it to be high-cardinality
+        //    based on the sdk information (see `set_default_transaction_source`).
+        let treat_as_url = matches!(
```

Comment:
nit: do you think extracting these conditions into a separate method would ease the readability of this method?

Reply:
Yep, done.